server/src/main/java/org/elasticsearch/index/mapper/MapperMergeContext.java

```diff
@@ -43,10 +52,57 @@ private MapperMergeContext(MapperBuilderContext mapperBuilderContext) {
      * @return a new {@link MapperMergeContext} with this context as its parent
      */
     public MapperMergeContext createChildContext(String name) {
-        return new MapperMergeContext(mapperBuilderContext.createChildContext(name));
+        return createChildContext(mapperBuilderContext.createChildContext(name));
+    }
+
+    /**
+     * Creates a new {@link MapperMergeContext} with a given {@link MapperBuilderContext}
+     * @param childContext the child {@link MapperBuilderContext}
+     * @return a new {@link MapperMergeContext}, wrapping the provided {@link MapperBuilderContext}
+     */
+    public MapperMergeContext createChildContext(MapperBuilderContext childContext) {
+        return new MapperMergeContext(childContext, remainingFieldsUntilLimit);
     }
 
     MapperBuilderContext getMapperBuilderContext() {
         return mapperBuilderContext;
     }
+
+    void removeRuntimeField(Map<String, RuntimeField> runtimeFields, String name) {
+        if (runtimeFields.containsKey(name)) {
+            runtimeFields.remove(name);
+            if (remainingFieldsUntilLimit.get() != Long.MAX_VALUE) {
+                remainingFieldsUntilLimit.incrementAndGet();
+            }
+        }
+    }
+
+    void addRuntimeFieldIfPossible(Map<String, RuntimeField> runtimeFields, RuntimeField runtimeField) {
+        if (runtimeFields.containsKey(runtimeField.name())) {
+            runtimeFields.put(runtimeField.name(), runtimeField);
+        } else if (canAddField(1)) {
+            remainingFieldsUntilLimit.decrementAndGet();
+            runtimeFields.put(runtimeField.name(), runtimeField);
+        }
```

Comment:
yea I guess I forgot the canAddField call, but the point is that we can potentially call put in one place. It's not super important, I just thought it would make the code a tiny bit clearer, maybe it does not make a difference.

Reply:
Not sure if that's feasible as we're not calling `put` in all branches. In the `else` case (trying to add a new runtime field when the field limit is reached), we don't call `put`. However, this method is now removed anyway, resolving.